### ISSUE TITLE
Volume order in structure

### ISSUE THREAD

**Issue by Zelenyy** [mipt-npm/gdml.kt#5]
Package don't make true order of volume structure.
Code:
```
            structure {
                val matref = GDMLRef<GDMLMaterial>("G4_AIR")
                val worldBox = this@GDML.solids.box("World", 100, 100, 100)
                val volWorld = volume("world", matref, worldBox.ref()){}
                val segment = this@GDML.solids.tube("InnerTube", 20, 5.0){
                    rmin = 17
                    deltaphi = 60
                    aunit = "degree"
                }
                val vol = volume("vol1", matref, segment.ref()) {}
                this@GDML.setup.world = volWorld.ref()
                this@GDML.world.apply {
                    physVolume(vol.ref()) {
                        positionref = ref("box_position")
                        rotationref = ref("Rot1")
                    }
                }
            }
```
Result --- volume `vol1` define after its using:
```
   <structure>
        <volume name="world">
            <physvol>
                <volumeref ref="vol1"></volumeref>
                <positionref ref="box_position"></positionref>
                <rotationref ref="Rot1"></rotationref>
            </physvol>
            <materialref ref="G4_AIR"></materialref>
            <solidref ref="World"></solidref>
        </volume>
        <volume name="vol1"> 
            <materialref ref="G4_AIR"></materialref>
            <solidref ref="InnerTube"></solidref>
        </volume>
    </structure>
```

In my gdml dsl i sorted tag by its children:
```
    override fun compareTo(other: VOLUME): Int {
        if (dependenciesVolumeList.contains(other.name)){
            return 1
        }
        if (other.dependenciesVolumeList.contains(this.name)){
            return -1
        }
        return 0
    }
```

**Comment by altavir** [comment]
The usage of `world` is no correct in generator anyway. It is meant for access only. I will create a separate issue for that.

As for the rest, it is possible to add additional sorting operation before conversion to string. For large files, it could be expensive though.

**Comment by Zelenyy** [comment]
> The usage of world is no correct in generator anyway. 
It does not matter, I think this bug will reproduction for any volume if his usage will be split on several part.

**Comment by Zelenyy** [comment]
>`World` should not be used in generator

How to set world volume?

**Comment by altavir** [comment]
World is defined here: https://github.com/mipt-npm/gdml.kt/blob/4369ac0477f0bb6eaede6d09fae068b0c242220a/src/commonMain/kotlin/scientifik/gdml/GDML.kt#L45. It uses the one in setup, otherwise takes first declared volume in structure. The second clause could create some problems. I've opened #6 for that.

**Comment by altavir** [comment]
> > The usage of world is no correct in generator anyway.
> 
> It does not matter, I think this bug will reproduction for any volume if his usage will be split on several part.

It is not a bug per se. The problem is that you say that GEANT does not read the whole structure before starting to interpret it. So the behavior needs to be adjusted.
